codec/codec-sofa-hessian/src/main/java/com/alipay/sofa/rpc/codec/sofahessian/SingleClassLoaderSofaSerializerFactory.java

```diff
@@ -73,12 +83,33 @@ public Deserializer getDeserializer(String type) throws HessianProtocolException
             Deserializer subDeserializer = getDeserializer(type.substring(1));
             deserializer = new ArrayDeserializer(subDeserializer);
         } else {
+            ClassLoader appClassLoader = Thread.currentThread().getContextClassLoader();
             try {
-                ClassLoader appClassLoader = Thread.currentThread().getContextClassLoader();
+                if (!dynamicLoadEnable) {
+                    Map<String, Object> typeMap = _typeNotFoundMap.get(appClassLoader);
+                    if (typeMap != null) {
+                        if (typeMap.containsKey(type)) {
+                            return null;
+                        }
+                    }
+                }
                 Class<?> cl = Class.forName(type, true, appClassLoader);
                 deserializer = getDeserializer(cl);
             } catch (Exception e) {
                 if (e instanceof ClassNotFoundException) {
+                    if (!dynamicLoadEnable) {
+                        Map<String, Object> typeMap = _typeNotFoundMap.get(appClassLoader);
```

Comment:
_typeNotFoundMap is a ConcurrentHashMap, you can use `computeIfAbsent` method to avoid using synchronized block.

Reply:
resolved and add tests